e2e/test/iothub/service/ServiceClientE2ETests.cs

```diff
@@ -63,5 +64,71 @@ private async Task TestTimeout(TimeSpan? timeout)
                 Logger.Trace($"Testing ServiceClient SendAsync(): exiting test after time={sw.Elapsed}; ticks={sw.ElapsedTicks}");
             }
         }
+
+        [LoggedTestMethod]
+        public async Task MessageIdDefaultNotSet_SendEventDoesNotSetMessageId()
```

Comment:
Unfortunately the way `AmqpServiceClient` is implemented, it makes mocking the required amqp types difficult (the amqp types are private members of the class, and cannot be set from any public/ internal API). This is making mocking the amqp behavior difficult.
To get by that, I am testing this in the E2E flow, even though this is a unit test scenario.